src/cpp/rtps/persistence/SQLite3PersistenceService.cpp

```diff
@@ -29,29 +29,138 @@ namespace eprosima {
 namespace fastrtps{
 namespace rtps {
 
-static sqlite3* open_or_create_database(const char* filename)
+static int database_version(sqlite3* db)
+{
+    sqlite3_stmt* version_stmt;
+    int version = 0;
+    sqlite3_prepare_v2(db, "PRAGMA user_version;", -1, &version_stmt, NULL);
+    while (SQLITE_ROW == sqlite3_step(version_stmt))
```

Comment:
The sentence `PRAGMA user_version;` should return a single result, I'd change `while` with `if`.